extension.js

```diff
@@ -912,17 +913,48 @@ function handle_doc_open(active_doc) {
 
 function quote_field(field, delim) {
     if (field.indexOf('"') != -1 || field.indexOf(delim) != -1) {
-        return '"' + field.replace(/"/g, '""')  + '"';
+        return '"' + field.replace(/"/g, '""') + '"';
     }
     return field;
 }
 
 
 function quoted_join(fields, delim) {
-    var quoted_fields = fields.map(function(val) { return quote_field(val, delim); });
+    var quoted_fields = fields.map(function (val) { return quote_field(val, delim); });
     return quoted_fields.join(delim);
 }
 
+function sample(uri) {
+    var filePath = uri.fsPath;
+
+    var sizeLimit = 102400; // 100 KB
+    var fileSizeInBytes = fs.statSync(filePath)['size'];
+    if (fileSizeInBytes < sizeLimit) {
+        vscode.window.showInformationMessage('No need to cut.')
+        return;
+    }
+
+    var rl = readline.createInterface({
```

Comment:
I am not familiar with node, so correct me if I am wrong.

I searched and found this node built-in module and [the example](https://nodejs.org/api/readline.html#readline_example_read_file_stream_line_by_line) from the docs.

And for the `fs.read()`, I don't see any benefit. We need to `fs.open()`, `fs.read()`, get the `Buffer`, `toString()` and `split(/\r?\n/)`. Why bother doing these given node already provides a official module 😳

Reply:
You don't have to split after reading with `fs.read()`, just use `let pos = lastIndexOf('\n')`, drop everything after `pos`, and write the remaining string to dst. The problem with readline, is that you have to use `lines.join(os.EOL)` at the end, and this can change file content, e.g. original line separator was `'\n'` but `os.EOL` on Win would be `\r\n`. Also fs.read() should be faster.

Reply:
Makes sense to me 👍

Reply:
=) BTW another advantage is that we would have to use fs.read() anyway to implement "tail" since it has `position` argument to start read from. With readline we would have to read through the whole file before getting to the tail part.

Reply:
I tried

```js
var buffer = Buffer.from([]);
fs.read(fd, buffer, 0, size_limit, 0, function(err, num) {
    if (err) {
        console.log(err);
        return;
    }
    ...
});
```

but received

```
Error: Offset is out of bounds
	at Object.fs.read (fs.js:664:11)
	at fs.open
```

I am confused. How can offset be out of bounds since it is 0? 😂